File: PRACTICA/INTRODUCCION A POO/POO9/computadora.py
class Computadora:
    def __init__(self, procesador, ram, almacenamiento):
        self.procesador = procesador
        self.ram = ram
        self.almacenamiento = almacenamiento
        self.estado = "Apagada"  
    
    # Método para mostrar los componentes principales
    def mostrar_componentes(self):
        print(f"Procesador: {self.procesador}")
        print(f"Memoria RAM: {self.ram} GB")
        print(f"Almacenamiento: {self.almacenamiento} GB")
    
    # Método para mostrar el estado de la computadora
    def mostrar_estado(self):
        print(f"La computadora está {self.estado}.")
    
    # Método para encender la computadora
    def encender(self):
        if self.estado == "Apagada":
            self.estado = "Encendida"
            print("La computadora ha sido encendida.")
        else:
            print("La computadora ya está encendida.")
    
    # Método para apagar la computadora
    def apagar(self):
        if self.estado == "Encendida":
            self.estado = "Apagada"
            print("La computadora ha sido apagada.")
        else:
            print("La computadora ya está apagada.")

comp = Computadora("Intel Core i7", 16, 512)

comp.mostrar_componentes()
comp.mostrar_estado()
comp.encender()  
comp.mostrar_estado()  

comp.apagar() 
comp.mostrar_estado() 

File: PRACTICA/INTRODUCCION A POO/POO9/ejercicio4/Computadora.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package ejercicio4;

/**
 *
 * @author Administrador
 */
class Computadora {
    private String procesador;
    private int memoriaRAM;
    private int almacenamiento;
    private boolean encendida;

    public Computadora(String procesador, int memoriaRAM, int almacenamiento) {
        this.procesador = procesador;
        this.memoriaRAM = memoriaRAM;
        this.almacenamiento = almacenamiento;
        this.encendida = false;
    }
    //B) Muestra el estado de la computadora (encendido o apagado)
    public void encender() {
        encendida = true;
    }

    public void apagar() {
        encendida = false;
    }
    //a) Muestra los componentes principales de la computadora
    public void mostrar() {
        System.out.println("Procesador: " + procesador+" Memoria RAM: " + memoriaRAM+" Almacenamiento: " + almacenamiento );
        System.out.println("Estado: ");
        if(encendida==true){
            System.out.println("\tEncendida");
        }
        else{
             System.out.println("\tApagada");
        }
    }
}
